ex7/t_hayakawa/main.py

```diff
@@ -0,0 +1,267 @@
+import numpy as np
+import argparse
+import time
+import pandas as pd
+import matplotlib.pyplot as plt
+from scipy.stats import norm
+from sklearn.cluster import KMeans
+
+
+class GMM:
+    def __init__(self, data, K, mu, sigma, pi) -> None:
```

Comment:
ここやほかの関数について、各データのshapeを定義するdocstringがあると親切かと思います。
```suggestion
    def __init__(self, data, K, mu, sigma, pi) -> None:
        """
        # Args:
            data (ndarray, shape=(N,D)):  training data
            K (...
        """
```